[Jetpack Performance] Refresh site status periodically

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
Closes WOOMOB-1182

### Description
This PR adds logic to make sure we refresh site status periodically, it implements the approach suggested here pe5sF9-4ye-p2#comment-4801:
- Store the timestamp when flagging a website as not supporting app passwords.
- When the timestamp is older than 7 days, we ignore it.

### Steps to reproduce
Follow the TC7 test steps from here pe5sF9-4Am-p2

### Testing information
- Confirm that after the duration, we check the site again, and we can use app passwords.

### The tests that have been performed
The above.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Handle clearing flags after a given duration

hichamboushaba committed:
Synchronize functions to avoid issues on race conditions

hichamboushaba committed:
Add unit tests

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba commented on libs/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpapi/applicationpasswords/JetpackApplicationPasswordsSupport.kt:
```diff
@@ -1,28 +1,72 @@
     companion object {
-        private const val UNSUPPORTED_JETPACK_APP_PASSWORDS_SITES = "unsupported_jetpack_app_passwords_sites"
+        @VisibleForTesting
+        internal const val FLAG_REFRESH_DURATION_DAYS = 7
```
The duration is not confirmed yet, I will confirm it with Huong, and update if needed in a separate PR.

hichamboushaba replied:
We agreed to increase this to 14 days pe5sF9-4ye-p2#comment-4815, this was implemented in 7e3bc60

hichamboushaba commented on libs/fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpapi/applicationpasswords/JetpackApplicationPasswordsSupport.kt:
```diff
@@ -1,28 +1,72 @@
+            ?.firstOrNull { it.startsWith("${siteModel.siteId}:") }
+
+        if (siteFlag != null) {
+            val flagTimestamp = siteFlag.substringAfter(":").toLongOrNull() ?: 0L
+            if (System.currentTimeMillis() - flagTimestamp < FLAG_REFRESH_DURATION_DAYS.days.inWholeMilliseconds) {
```
I think using `0L` in this case is still accepted, we just want to handle migration for entries added without the timestamp, using `0L` will cause the flag to be removed in next lines.

hichamboushaba committed:
Increase the refresh duration to 14 days

hichamboushaba committed:
Fix test case

JorgeMucientes reviewed (approved):
Good job! It works as expected @hichamboushaba. 

However, one remark about the testing instructions for our future selves when beta testing this: 
Instructions tell to test TC7 from pe5sF9-4Am-p2. TC7 point to use TC3 snippet in order to mark the site as application passwords unsupported. However, with the current logic and the provided snippet in TC3 the site won't be flagged as unsupported because the API error returned with TC3 snippet is `application_passwords_disabled` whereas the app expects `application_passwords_disabled_for_user`: 

<img width="992" height="311" alt="Image" src="https://github.com/user-attachments/assets/c0748739-895e-45f1-a7a8-1a0a6b5d2e24" />

In the real case scenario, what would WP API return? `application_passwords_disabled_for_user` or `application_passwords_disabled`. If the former, then we just need to update the TC3 snippet to fix the testing instructions; if the latter, then we should fix the issue in the code asap. Also why not match agains the `httpStatusCode` 501 instead of matching against `apiErrorCode` ?

hichamboushaba commented:
Good eye @JorgeMucientes, yes this is an issue, I already fixed it in my next PR in this commit https://github.com/woocommerce/woocommerce-android/commit/81de96d7b7ad3f4098e6448760b303b9f4e05038, and updated the error handling plan to include both error codes: pe5sF9-4ye-p2#comment-4787
